client/state/support/reducer.js

```diff
@@ -1,11 +1,39 @@
 /** @format */
 
 /**
- * External dependencies
+ * Internal dependencies
  */
+import { createReducer, combineReducers } from 'state/utils';
+import {
+	SUPPORT_SESSION_ACTIVATE,
+	SUPPORT_SESSION_EXPIRE,
+	SUPPORT_USER_ACTIVATE,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
 
-import { combineReducers } from 'state/utils';
-import { SUPPORT_USER_ACTIVATE } from 'state/action-types';
+export const SUPPORT_SESSION_STATE = {
+	NONE: 'none',
+	ACTIVE: 'active',
+	EXPIRED: 'expired',
+};
+
+export const supportSession = createReducer( SUPPORT_SESSION_STATE.NONE, {
+	[SUPPORT_SESSION_ACTIVATE]( state ) {
+		return state === SUPPORT_SESSION_STATE.NONE ? SUPPORT_SESSION_STATE.ACTIVE : state;
+	},
+	[SUPPORT_SESSION_EXPIRE]( state ) {
+		return state === SUPPORT_SESSION_STATE.ACTIVE ? SUPPORT_SESSION_STATE.EXPIRED : state;
+	},
+	// Use custom serialization so we're able to relay supportSession state from server to client.
+	// Otherwise, only the default state is used when deserializing redux state from the server.
+	[SERIALIZE]() {
+		return null;
+	},
+	[DESERIALIZE]( state ) {
+		return state === null ? SUPPORT_SESSION_STATE.NONE : state;
+	},
+} );
```

Comment:
did you have something you wanted to achieve by using `createReducer()`? if not then a simpler single function might do just fine.

I'm going to toss some ideas out here too that I'll add in another comment but I wonder if we need separate `_ACTIVATE` and `_EXPIRE` actions or if we could get by with a simpler single action to set the state…

(note the use of our helper `withoutPersistence()` introduced in #14289)

```js
export const supportSession = withoutPersistence(
	( state = SUPPORT_SESSION_STATE.NONE, { type, nextState } ) =>
		SUPPORT_SESSION_TOGGLE === type
			? nextState
			: state
);
```

Reply:
The use of `createReducer` is sort of a hack because it was the combination of things I could find that would result in `support.supportSession` being received from the server-provided state.

If we use `withoutPersistence`, then the server-provided value is thrown away due to [initial state deserialization](https://github.com/Automattic/wp-calypso/blob/master/client/state/initial-state.js#L45) and [withoutPersistence deserialization](https://github.com/Automattic/wp-calypso/blob/master/client/state/utils.js#L371) which ignores the state passed to the reducer. 

We don't want to actually persist `support.supportSession` but want to convey its value from the server to the client. My current approach is to mark it for persistence but to not actually persist its value. What would be a better approach here?

Reply:
> the combination of things I could find that would result in support.supportSession being received by the client from the server-provided state

if that's the case then I think `hasCustomPersistence` will do you without bringing in the complexity of `createReducer`, plus you can stick to simpler constructs

```js
export const supportSession = ( state = NO_SESSION, { type } ) => {
	switch ( type ) {
		case SUPPORT_SESSION_ACTIVATE:
			return state === NO_SESSION ? ACTIVE : state;

		case SUPPORT_SESSION_EXPIRE:
			return state === ACTIVE ? EXPIRED : state;

		case SERIALIZE:
			return null;

		case DESERIALIZE:
			return state === null ? NO_SESSION : state;

		default:
			return state;
	}
}

supportSession.hasCustomPersistence = true;
```